### DISCUSSION TITLE
How to backup and restore just the volume contents?

### DISCUSSION THREAD

**Discussion by Cavaler** [vmware-tanzu/velero#8445] [Community support Q&A] [answered]
Some background: we are looking for a solution to help us upgrade clusters from quite old k3s and Longhorn versions. In-place upgrade was deemed too complicated and didn't work. So what we are trying to do is back up all the data, and recreate the cluster from scratch, with newer versions of k3s, Longhorn, and also our own Helm charts.

But we also want to restore the data in both the Longhorn and local-path volumes. Is there a way to do that using velero? Especially in the situation when the resources themselves are first created by Helm charts? In the latter case there is specifically a note saying that during update " In case of PVCs for example, data in the PV is not restored or overwritten in any way"

**Comment by kaovilai** [comment, marked as answer]
> data in the PV is not restored or overwritten in any way

is how it is in velero. There is no flag that will trigger volume content override. You can force restore by deleting PVC prior to `velero restore`.

track request here: https://github.com/vmware-tanzu/velero/issues/7345  and throw a 👍
